backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala

```diff
@@ -231,21 +238,22 @@ private final class BloopNameHashing(
   }
 
   def recompileClasses(
-      sources: Set[VirtualFile],
+      currentInvalidatedSources: Set[VirtualFile],
```

Comment:
This is the main change, we would previously send all invalidated files even if they were no longer contained in the sources. This make sense for class files to remove them, but no some much to compile new files.

With this change I was no longer able to reproduce the issue in Metals where we would get all compilation artifacts somehow.